tl

```diff
@@ -42,25 +42,40 @@ end
 -- FIXME
 local function validate_config(config)
    local valid_keys = {
-      build_dir = true,
-      exclude = true,
-      files = true,
-      include = true,
-      include_dir = true,
-      preload_modules = true,
-      quiet = true,
-      source_dir = true,
-      skip_compat53 = true,
+      build_dir = "string",
+      exclude = "{string}",
+      files = "{string}",
+      include = "{string}",
+      include_dir = "{string}",
+      preload_modules = "{string}",
+      quiet = "boolean",
+      source_dir = "string",
+      skip_compat53 = "boolean",
    }
 
-   for k, _ in pairs(config) do
+   for k, v in pairs(config) do
       if not valid_keys[k] then
          print(string.format("Warning: unknown key '%s' in tlconfig.lua", k))
+      else
+         -- TODO: could we type-check the config file using tl?
```

Comment:
Good question! Probably, at some point! But this PR is already a good improvement!!